ksql-execution/src/main/java/io/confluent/ksql/execution/plan/StreamSource.java

```diff
@@ -63,22 +98,315 @@ public boolean equals(final Object o) {
+      this.timestampTransformer
+          = Objects.requireNonNull(timestampTransformer, "timestampTransformer");
+      this.consumedFactory = Objects.requireNonNull(consumedFactory, "consumedFactory");
+      this.physicalSchemaFactory
+          = Objects.requireNonNull(physicalSchemaFactory, "physicalSchemaFactory");
+    }
+  }
+
+  static StreamSource<Windowed<Struct>> createWindowed(
+      final QueryContext queryContext,
+      final LogicalSchemaWithMetaAndKeyFields schemaWithMetaAndKeyFields,
+      final String topicName,
+      final KeyFormat keyFormat,
+      final ValueFormat valueFormat,
+      final Set<SerdeOption> options,
+      final TimestampExtractionPolicy timestampPolicy,
+      final int timestampIndex,
+      final Optional<AutoOffsetReset> offsetReset,
+      final Constructor<Windowed<Struct>> constructor
+  ) {
+    return constructor.construct(
+        new DefaultExecutionStepProperties(
+            queryContext.toString(),
+            schemaWithMetaAndKeyFields.getSchema(),
+            queryContext
+        ),
+        topicName,
+        Formats.of(keyFormat, valueFormat, options),
+        timestampPolicy,
+        timestampIndex,
+        offsetReset,
+        schemaWithMetaAndKeyFields.getOriginalSchema(),
+        new Functions<>(
+            (fmt, schema, ctx, builder) -> builder.buildKeySerde(
+                fmt.getFormatInfo(),
+                fmt.getWindowInfo().get(),
+                schema,
+                ctx
+            ),
+            windowedMapper(schemaWithMetaAndKeyFields.getSchema()),
+            new AddTimestampColumn(),
+            Consumed::with,
+            PhysicalSchema::from
+        )
+    );
+  }
+
+  public static StreamSource<Windowed<Struct>> createWindowed(
+      final QueryContext queryContext,
+      final LogicalSchemaWithMetaAndKeyFields schemaWithMetaAndKeyFields,
+      final String topicName,
+      final KeyFormat keyFormat,
+      final ValueFormat valueFormat,
+      final Set<SerdeOption> options,
+      final TimestampExtractionPolicy timestampPolicy,
+      final int timestampIndex,
+      final Optional<AutoOffsetReset> offsetReset) {
+    return createWindowed(
+        queryContext,
+        schemaWithMetaAndKeyFields,
+        topicName,
+        keyFormat,
+        valueFormat,
+        options,
+        timestampPolicy,
+        timestampIndex,
+        offsetReset,
+        StreamSource::new
+    );
+  }
+
+  static StreamSource<Struct> createNonWindowed(
+      final QueryContext queryContext,
+      final LogicalSchemaWithMetaAndKeyFields schemaWithMetaAndKeyFields,
+      final String topicName,
+      final KeyFormat keyFormat,
+      final ValueFormat valueFormat,
+      final Set<SerdeOption> options,
+      final TimestampExtractionPolicy timestampPolicy,
+      final int timestampIndex,
+      final Optional<AutoOffsetReset> offsetReset,
+      final Constructor<Struct> constructor
+  ) {
+    return constructor.construct(
```

Comment:
The main reason for this was to get to the various lambdas we pass to streams so that those can be tested. Now that I think about it though we can capture those from the streams mocks. Let me try that.